### ISSUE TITLE
Fix level manager, we have to talk

### ISSUE THREAD

**Issue by TizioMaurizio** [VDP-noclip/noclip#104]
>https://github.com/VDP-noclip/noclip/issues/55
  
I quote the above issue to highlight the fact that we created a Level Manager that is all but fool proof, and I invite you to try it to prove me wrong.  
  
The issue in question was created by me to complete the work on the level manager I was developing, unfortunately it was completely disregarded and it was decided to redesign and remake everything halfway through without even checking the work that was already there, and the decisions that started this complete redesign were made without prior knowledge nor experience on how to build a level for our game.  
  
I would like to remind that we split the work beforehand, and we agreed that I would work on the level manager, so anyone that wanted to contribute was supposed to come to me for a briefing on the existing work.  
  
In the name of teamwork I decided to implement anyway a structure I was strongly against, throwing away my work, and I did it exactly as we specified to do it, in fact it is working flawlessly _at runtime_.  
  
As it was my thought such structure is being proved inappropriate for the developement of our game (and I would argue for any game) and very hard to deal with while creating levels, in few words: sounds good, doesn't work.  
  
I already tested and discussed about the Level manager I originally planned with our level designer, and without extra steps, using that one, it is possible to play the scene from the editor without having to drag prefabs and scenes all over unity, something very simple but for which we have to crave right now.  
  
This issue is here just to discuss the coming update on the Level Manager and see if anything will need to be accustomed, honestly it is such a simple structure that I hardly believe anything will suffer side effects apart from level managing related stuff.  
  
.  
  
In conclusion: the assumption that was made to complicate so much such a simple task was that more people would work in parallel on one Area, honestly from here to the game delivery I see hardly one person working on one area at a specific time, which is ok because we are not building a triple A.

**Comment by TizioMaurizio** [comment]
![red-heart_2764-fe0f](https://user-images.githubusercontent.com/48728364/203584314-ac43b527-1cbf-40f4-8718-0b5269b7b561.png)
I still love you all, the only person I'm angry with is myself for not insisting enough on something I knew I was right about

**Comment by TizioMaurizio** [comment]
https://en.wikipedia.org/wiki/Rube_Goldberg_machine
![Rube](https://user-images.githubusercontent.com/48728364/203612063-ce84749e-e769-4aef-aafc-2125112c89ac.jpg)

**Comment by stefanofossati** [comment]
Honestly I haven't developed a complete level with all the puzzles so I I haven't taken in consideration the working on Unity that obliges to drag and drop everything. We have to rediscuss everything taking in consideration the your and @AleDeNic  feedbacks so that we can develop a more manageble area manager. 
Maybe we can write here all the consideration and discuss them together in presence.

In any case I think that we are working well, so I'm convinced that we can create a very good structure under all the aspects.  :)
